apps/internal/oauth/ops/authority/authority.go

```diff
@@ -138,6 +138,38 @@ const (
 	ADFS = "ADFS"
 )
 
+type AuthenticationScheme interface {
```

Comment:
Just a reminder that all interface changes are breaking--if there's a possibility we'll want to add, remove, or tweak some aspect of this in the future, it shouldn't be an interface

Reply:
@manojampalam

Reply:
good point. @bgavrilMS your thoughts?  Equivalent interface in .Net is [internal ](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/c89dd200d3c1e4f6f171106403509d685dec01e0/src/client/Microsoft.Identity.Client/AuthScheme/IAuthenticationScheme.cs#L12) at this point, as PoP support is built in via its SHR feature. 

Otherwise, I feel, this interface as such is sufficient for the purpose of client implemented PoP enablement. If at all we need more functionality in future, we could have it via a new option based on AuthenticationSchemeV2 ?

Reply:
This is one of the reasons why we put in the disclaimer that this API is an extensbility point created for a very specific scenario. In the off-chance we need to modify this interface, we could either go via the `AuthenticationScheme2` route or  introduce a breaking change in an API that isn't meant for wide consumption. We have done this in the past in MSAL.NET and it worked ok. We can discuss in more detail and decide if a breaking change arises.